### PR TITLE
Test volume clerk client

### PR DESCRIPTION
**Pull request by dbox-fnal**

### CODE REVIEW

File: src/tests/test_volume_clerk_client.py [hunk 1]
```diff
@@ -0,0 +1,1301 @@
...
+        self.assertEqual(t, -1)
+
+    def test_capacity_str(self):
+        inp = 1024
+        c1 = volume_clerk_client.capacity_str(inp, 'B')
+        self.assertEqual(c1, "1024.00B ")
+        c2 = volume_clerk_client.capacity_str(inp, 'KB')
+        self.assertEqual(c2, "1024.00B ")
+        inp *= 1024
+        c3 = volume_clerk_client.capacity_str(inp)
```
Comment (renbauer-fnal) [inline src/tests/test_volume_clerk_client.py:50]: Can we add a test case for KB here and one that sends a negative?

File: src/tests/test_volume_clerk_client.py [hunk 2]
```diff
@@ -0,0 +1,1301 @@
...
+        cmp = "VR6995M8         8183.50GB   (['none', 'none'] "
+        cmp += "0101-0000 ['none', 'none'] 0101-1200)   LTO8F1 "
+        cmp += "          M8.volume_read_test.cpio_odc         "
+        cmp += "totally made up\n"
+
+        with mock.patch('sys.stdout', new=StringIO.StringIO()) as std_out:
+            volume_clerk_client.show_volume(fake_vol)
+            self.assertEqual(std_out.getvalue(), cmp)
+
+    def test_my_atol(self):
```
Comment (renbauer-fnal) [inline src/tests/test_volume_clerk_client.py:112]: Do we know what this function (my_atol) actually does? It appears to parse units to standardize values but what does "atol" mean? Why isn't it called my_size_in_bytes?

File: src/tests/test_volume_clerk_client.py [hunk 3]
```diff
@@ -0,0 +1,1301 @@
...
+
+    def test_set_system_full(self):
+        expected_work_ticket = {'work': 'set_system_full',
+                                'external_label': 'external_label'}
+        self.sent_msg.reset_mock()
+        self.vcc.set_system_full('external_label')
+        generated_work_ticket = self.sent_msg.mock_calls[0][1][0]
+        self.assertEqual(generated_work_ticket, expected_work_ticket)
+
+    def test_set_system_migrated(self):
```
Comment (renbauer-fnal) [inline src/tests/test_volume_clerk_client.py:393]: Man, all these are almost identical. Can we simplify these functions to be something like:
```
def assert_generates_work_ticket(work_str, test_func, extra_items={}):
  expected_work_ticket = {'work': work_str,
                          'external_label': 'external_label'} + extra_items
  self.sent_msg.reset_mock()
  test_func()
  generated_work_ticket = self.sent_msg.mock_calls[0][1][0]
  self.assertEqual(generated_work_ticket, expected_work_ticket)

def test_set_system_migrated(self):
  assert_generates_work_ticket('set_system_migrates', 
     lambda: self.vcc.set_system_migrated('external_label'))
```

Maybe there is a cleaner way than passing lambdas but my kwargs-fu is not up to it atm

File: src/tests/test_volume_clerk_client.py [hunk 4]
```diff
@@ -0,0 +1,1301 @@
...
+                            "Wrong number of arguments" in std_out.getvalue(),
+                            std_out.getvalue())
+                        self.vci.args = []
+
+                        # now test with some correct args
+                        sent_msg.reset_mock()
+                        std_out.truncate()
+                        self.vci.args.append('noaccess')
+                        volume_clerk_client.do_work(self.vci)
+                        arglebargle = """{'not': None, 'work': 'get_vols3', 'key': None, 'in_state': 'noaccess'}"""
```
Comment (renbauer-fnal) [inline src/tests/test_volume_clerk_client.py]: Nice var name, but is this used?

File: src/tests/test_volume_clerk_client.py [hunk 5]
```diff
@@ -0,0 +1,1301 @@
...
+                        self.assertTrue(
+                            """'work': 'get_vol_list2'""" in str(
+                                sent_msg.mock_calls), test + ': ' + str(
+                                sent_msg.mock_calls))
+
+                        # -------------
+                        test = "next"
+                        # -------------
+                        self.vci.next = 1
+                        self.vci.labels = 0
```
Comment (renbauer-fnal) [inline src/tests/test_volume_clerk_client.py]: Can we move these resets to the bottom of the test cases above, like you are doing later in these cases?

**Review by renbauer-fnal — CHANGES_REQUESTED** [review]

File: src/tests/test_volume_clerk_client.py (hunk 5 above)
Comment (dbox-fnal) [reply src/tests/test_volume_clerk_client.py]: Done.

File: src/tests/test_volume_clerk_client.py (hunk 4 above)
Comment (dbox-fnal) [reply src/tests/test_volume_clerk_client.py]: Must be a remnant from an earlier test approach, removing does not affect the test

File: src/tests/test_volume_clerk_client.py (hunk 3 above)
Comment (dbox-fnal) [reply src/tests/test_volume_clerk_client.py:393]: There are most likely ways to simplify this,  I will experiment on other repetitive client tests.  This one is done and works so I would prefer to leave it as is.  The test of do_work() lower down is also repetitive and ugly and I would like even more to come up with something parameterized for it.

File: src/tests/test_volume_clerk_client.py (hunk 2 above)
Comment (dbox-fnal) [reply src/tests/test_volume_clerk_client.py:112]: My guess is named after atol(3) in the standard C library.

**Comment by dbox-fnal** [comment]
reviewer suggestions implemented, waiting for approval

**Review by renbauer-fnal — APPROVED** [review]
